pkg/kwok/server/metrics.go

```diff
@@ -17,11 +17,49 @@ limitations under the License.
 package server
 
 import (
-	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
+	"sigs.k8s.io/kwok/pkg/kwok/metrics"
 )
 
 // InstallMetrics registers the metrics handler on the given mux.
-func (s *Server) InstallMetrics() {
-	promHandler := promhttp.Handler()
-	s.restfulCont.Handle("/metrics", promHandler)
+func (s *Server) InstallMetrics() error {
+	defaultMetrics := metrics.DefaultMetrics()
+	s.restfulCont.Handle("/metrics", defaultMetrics.Handler())
+	for _, m := range s.config.Metrics {
+		path, err := executePathTemplate(m.Spec.Path, m)
+		if err != nil {
+			return err
```

Comment:
What should we do in case path template is incorrect? Log the warning and continue, or stop the installation?